### PR TITLE
Add dd mm yyyy for hi

### PR DESCRIPTION
**Pull request by ruthvik-17**
added _gregorian_day_month_year_format to indic languages.

### CODE REVIEW

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/tests/temporal/date/en/test_date_detection.py [hunk 1]
```diff
@@ -244,4 +246,31 @@ def test_en_gregorian_year_day_month_format(self):
...
+        message = u'१/३/६६'
+        locale = 'hi-in'
+        # If we run
+        day1 = 1
+        month = 3
+        year1 = 1966
+        bot_message = u'जन्मदिन'
+
+        date_detector_object = DateAdvancedDetector(entity_name=self.entity_name, language='hi', locale=locale)
+        date_detection.set_bot_message(bot_message=parameters_dict[PARAMETER_BOT_MESSAGE])
```
Comment (srajdev) [inline ner_v2/tests/temporal/date/en/test_date_detection.py]: F821 undefined name 'date_detection'
F821 undefined name 'parameters_dict'
F821 undefined name 'PARAMETER_BOT_MESSAGE'

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 2]
```diff
@@ -1999,7 +2000,8 @@ def normalize_year(self, year):
         this_century = int(str(self.now_date.year)[:2])
         if len(year) == 2:
             if self.bot_message:
-                if past_regex and past_regex.search(self.bot_message):
+                if (self.past_date_referenced is True) or (past_regex and past_regex.search(self.bot_message)
+                                                           and int(year) > int(str(self.now_date.year)[2:])):
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: W503 line break before binary operator

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 3]
```diff
@@ -537,6 +538,120 @@ def _detect_weekday(self, date_list, original_list):
...
+        # past_regex = re.compile(ur'birth|bday|dob|born|जन्म|जन्मदिन|పుట్టినరోజు|పుట్టిన', flags=re.UNICODE)
+        past_regex = None
+        # Todo: Add more language variations of birthday.
+        present_regex = None
+        future_regex = None
+        this_century = int(str(self.now_date.year)[:2])
+        if len(year) == 2:
+            if self.bot_message:
+                if self.is_past_referenced or (past_regex and past_regex.search(self.bot_message)
+                                               and int(year) > int(str(self.now_date.year)[2:])):
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: W503 line break before binary operator

File: ner_v2/tests/temporal/date/en/test_date_detection.py [hunk 4]
```diff
@@ -244,4 +246,31 @@ def test_en_gregorian_year_day_month_format(self):
...
+        message = u'१/३/६६'
+        locale = 'hi-in'
+        # If we run
+        day1 = 1
+        month = 3
+        year1 = 2066
+        past_date_referenced = True
+
+        date_detector_object = DateAdvancedDetector(entity_name=self.entity_name, language='hi', locale=locale,
+                                                    past_date_referenced= past_date_referenced)
```
Comment (srajdev) [inline ner_v2/tests/temporal/date/en/test_date_detection.py]: E251 unexpected spaces around keyword / parameter equals

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 5]
```diff
@@ -537,6 +538,120 @@ def _detect_weekday(self, date_list, original_list):
...
+        # past_regex = re.compile(ur'birth|bday|dob|born|जन्म|जन्मदिन|పుట్టినరోజు|పుట్టిన', flags=re.UNICODE)
+        past_regex = None
+        # Todo: Add more language variations of birthday.
+        present_regex = None
+        future_regex = None
+        this_century = int(str(self.now_date.year)[:2])
+        if len(year) == 2:
+            if self.bot_message:
+                if self.past_date_referenced or (past_regex and past_regex.search(self.bot_message)
+                                                 and int(year) > int(str(self.now_date.year)[2:])):
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: W503 line break before binary operator

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 6]
```diff
@@ -1998,13 +1999,13 @@ def normalize_year(self, year):
...
         if len(year) == 2:
-            if self.bot_message:
-                if past_regex and past_regex.search(self.bot_message):
-                    return str(this_century - 1) + year
-                elif present_regex and present_regex.search(self.bot_message):
-                    return str(this_century) + year
-                elif future_regex and future_regex.search(self.bot_message):
-                    return str(this_century + 1) + year
+            if (((self.bot_message and past_regex.search(self.bot_message)) or (self.past_date_referenced is True))
+                    and (int(year) > int(str(self.now_date.year)[2:]))):
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: W503 line break before binary operator

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**